src/Tui.hs

```diff
@@ -103,18 +103,21 @@ msgSchema = MessageSchema { msgSchemaName = MN "SearchRequest"
 
 data FieldChoice = FReq | FOpt | FRep | Wrong deriving (Show, Read)
 
-toFC :: String -> FieldChoice
-toFC s = case s of
-    "query"    -> FReq
-    "maybeNum" -> FOpt
-    "numList"  -> FRep
-    _          -> Wrong
+toFC :: MessageSchema -> FieldName -> FieldChoice
+toFC (MessageSchema _ []) _ = Wrong
+toFC sch@(MessageSchema _ (SchemaField sfr _ sfn : sFields)) fn =
+    if fn == sfn
+    then case sfr of
+        SReq -> FReq
+        SOpt -> FOpt
+        SRep -> FRep
```

Comment:
If you just reuse value of `SchemaRule` data type then it probably makes sense for `toFC` function return `Either String SchemaRule` instead of `FieldChoice` and you can reuse existing type without introducing new data types. This also will make architecture design more consistent.